modules/tinymce/src/plugins/directionality/test/ts/browser/DirectionStyleTest.ts

```diff
@@ -0,0 +1,271 @@
+import { context, describe, it } from '@ephox/bedrock-client';
+import { TinyAssertions, TinyHooks, TinySelections, TinyUiActions } from '@ephox/wrap-mcagar';
+
+import Editor from 'tinymce/core/api/Editor';
+import Plugin from 'tinymce/plugins/directionality/Plugin';
+
+type Dir = 'rtl' | 'ltr';
+
+describe('browser.tinymce.plugins.directionality.DirectionStyleTest', () => {
+  const applyDir = (editor: Editor, dir: Dir) =>
+    TinyUiActions.clickOnToolbar(editor, dir === 'ltr' ? 'button[title="Left to right"]' : 'button[title="Right to left"]');
+
+  const testDirectionStyle = (editor: Editor, content: string, dir: Dir, expected: string) => {
+    editor.setContent(content);
+    TinySelections.setCursor(editor, [ 0, 0 ], 0);
+    applyDir(editor, dir);
+    TinyAssertions.assertContent(editor, expected);
+  };
+
+  const baseSettings = {
+    plugins: 'directionality',
+    toolbar: 'ltr rtl',
+    base_url: '/project/tinymce/js/tinymce'
+  };
+
+  context('TINY-9314: Inline direction style', () => {
+    const hook = TinyHooks.bddSetup<Editor>(baseSettings, [ Plugin ]);
+
+    it('Applying ltr with dir="rtl" and direction: rtl', () => testDirectionStyle(
+      hook.editor(),
+      '<p dir="rtl" style="direction: rtl;">Lorem ipsum</p>',
+      'ltr',
+      '<p style="direction: ltr;">Lorem ipsum</p>'
+    ));
+
+    it('Applying rtl with dir="ltr" and direction: ltr', () => testDirectionStyle(
+      hook.editor(),
+      '<p dir="ltr" style="direction: ltr;">Lorem ipsum</p>',
+      'rtl',
+      '<p dir="rtl" style="direction: rtl;">Lorem ipsum</p>'
```

Comment:
In the above `it` block the `dir` attribute is removed, but for this test case it isn't. I am curious what is causing that discrepancy. My guess it is the change here: https://github.com/tinymce/tinymce/pull/8690/files#diff-6399166c6f9c744f6fa49c73dfc28d1557920af20e49abdf0b842d8c7f40a455R28

I would expect using the ltr or rtl buttons to have equivalent HTML structure but maybe I am missing something.

Reply:
@ltrouton This is expected and aligns with previous behaviour. Since the parent (`body`) is in `ltr`, the `dir` attribute of the previous test was removed as it is unnecessary. In this case it is kept as the element has a different `dir` to its parent. The change you linked is a refactoring that should not change existing behaviour.